simulators/ethereum/pyspec/types.go

```diff
@@ -0,0 +1,280 @@
+package main
+
+import (
+	"encoding/json"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/common/math"
+	"github.com/ethereum/go-ethereum/core"
+	api "github.com/ethereum/go-ethereum/core/beacon"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+type testcase struct {
+	name       string
+	clientType string
+	fixture    fixtureTest
+	genesis    *core.Genesis
+	payloads   []*api.ExecutableData
+	postAlloc  core.GenesisAlloc
+	filepath   string
+}
+
+type fixtureTest struct {
+	json fixtureJSON
+}
+
+// UnmarshalJSON implements json.Unmarshaler interface.
+func (t *fixtureTest) UnmarshalJSON(in []byte) error {
+	if err := json.Unmarshal(in, &t.json); err != nil {
+		return err
+	}
+	return nil
+}
+
+type fixtureJSON struct {
+	Blocks     []block           	 `json:"blocks"`
+	Genesis    blockHeader           `json:"genesisBlockHeader"`
+	Pre        core.GenesisAlloc     `json:"pre"`
+	Post       core.GenesisAlloc     `json:"postState"`
+	BestBlock  common.UnprefixedHash `json:"lastblockhash"`
+	Network    string                `json:"network"`
+	SealEngine string                `json:"sealEngine"`
+}
+
+type block struct {
+	Rlp          string			     `json:"rlp"`
+	BlockHeader  *blockHeader        `json:"blockHeader"`
+	Transactions []transactions      `json:"transactions"`
+	UncleHeaders []byte              `json:"uncleHeaders"`
+	Withdrawals	 []withdrawals       `json:"withdrawals"`
+}
+
+type blockHeader struct {
+	ParentHash       common.Hash      `json:"parentHash"`
+	UncleHash        common.Hash      `json:"sha3Uncles"`
+	Coinbase         common.Address   `json:"miner"`
+	StateRoot        common.Hash      `json:"stateRoot"`
+	TransactionsTrie common.Hash      `json:"transactionsRoot"`
+	ReceiptTrie      common.Hash      `json:"receiptsRoot"`
+	Bloom            types.Bloom      `json:"bloom"`
+	Difficulty       *big.Int         `json:"difficulty`
+	Number           *big.Int         `json:"number"`
+	GasLimit         uint64           `json:"gasLimit"`
+	GasUsed          uint64           `json:"gasUsed"`
+	Timestamp        *big.Int         `json:"timestamp"`
+	ExtraData        []byte           `json:"extraData"`
+	MixHash          common.Hash      `json:"mixHash"`
+	Nonce            types.BlockNonce `json:"nonce"`
+	BaseFee          *big.Int         `json:"baseFeePerGas"`
+	Hash             common.Hash      `json:"hash"`
+	WithdrawalsRoot  common.Hash	  `json:"withdrawalsRoot`
+}
+
+func (b *blockHeader) UnmarshalJSON(input []byte) error {
+	type blockHeader struct {
```

Comment:
All `UnmarshalJSON` procedures in this file can be replaced by automatically-generated files by placing a comment like this:
```
//go:generate go run github.com/fjl/gencodec -type blockHeader -field-override blockHeaderMarshaling -out gen_bh.go
```
(where `blockHeaderMarshaling` is the structure contained in this procedure) somewhere within this file.

Then running `go generate` in the same folder as this file.

Make sure to use `go get github.com/fjl/gencodec` before running `go generate` for it to work properly.

Reply:
Changes here:
https://github.com/ethereum/hive/pull/732/commits/21148677b2cbfd86057d4c83ebdf704eef92650f#diff-5a6a26a4292c855637c91a0f97c3fae500808507cc0919c6172da75c8e748121